package/MDAnalysis/analysis/gnm.py

```diff
@@ -154,6 +158,45 @@ def order_list(w):
     return list_map
 
 
+def neighbour_generator(positions, cutoff):
```

Comment:
Now I have extracted the slow part in the GNM calculation into this function. The code duplication is also gone.  This function will give possible atom-pairs within a cutoff using a verlet-list avoiding having to compare all atom distances. I remember that @richardjgowers already has some cython code to quickly generate such a list. At a later state we can also think about moving this into `mda.lib` since it will likely be useful for others as well.